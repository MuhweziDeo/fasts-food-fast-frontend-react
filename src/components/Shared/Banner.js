import React from 'react';

const Banner=(props)=>{
    return(
        <section className="bg-primary">
        <div className="container py-md ">
            {props.children}
        </div>
        </section>

    );
};
 export default Banner;
