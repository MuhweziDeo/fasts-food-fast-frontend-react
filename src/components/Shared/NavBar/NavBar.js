import  React from 'react';
import { Dropdown, Image } from 'semantic-ui-react'
import {Link } from 'react-router-dom';

const trigger = (
    <span>
    <Image avatar src={localStorage.getItem('img')} /> {localStorage.getItem('username')}
  </span>
)

const options = [
    { key: 'user', text: 'Profile', icon: 'user', as:Link, to:'/dashboard'},
    { key: 'sign-out', text: 'Sign Out', icon: 'sign out', onClick:()=>{
        localStorage.clear()
            window.location.reload()
        } },
]

export const DropdownImageTrigger = () => (
    <Dropdown trigger={trigger} options={options} pointing='top left' icon={null} />
)
const NavBar = (props) =>{
    return(
        <nav className="navbar navbar-expand-lg navbar-dark bg-primary">

            <div className="container">
                <Link className="navbar-brand" to="/">Fasts Foods Fast</Link>
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbar-default"
                        aria-controls="navbar-default" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbar-default">
                    <div className="navbar-collapse-header">
                        <div className="row">
                            <div className="col-6 collapse-brand">
                                <a href="index.html">
                                    <img src="assets/img/brand/blue.png"/>
                                </a>
                            </div>
                            <div className="col-6 collapse-close">
                                <button type="button" className="navbar-toggler" data-toggle="collapse"
                                        data-target="#navbar-default" aria-controls="navbar-default"
                                        aria-expanded="false" aria-label="Toggle navigation">
                                    <span></span>
                                    <span></span>
                                </button>
                            </div>
                        </div>
                    </div>

                    <ul className="navbar-nav ml-lg-auto">
                        {!props.IsAuthenticated ?
                            <li className="nav-item  btn btn-white">
                                <Link to="/login"> Login </Link>

                            </li>
                            : <li className="nav-item text-white">
                            <DropdownImageTrigger/>
                        </li>}

                    </ul>

                </div>
            </div>
        </nav>
    )
};
export  default NavBar;
