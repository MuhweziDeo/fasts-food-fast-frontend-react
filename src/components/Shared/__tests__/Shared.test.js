import React from 'react';
import {shallow} from "enzyme/build";
import  NavBar from  '../NavBar/NavBar';
import Banner from '../Banner';
import DropdownImageTrigger from '../NavBar/NavBar';


describe('<Shared/>', () => {
    it('should render without crushing', () => {
        let wrapper = shallow(<NavBar/>);
        expect(wrapper).toMatchSnapshot()
    });
    it('should render without crushing', () => {
        let wrapper = shallow(<Banner/>);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render without crushing', () => {
        let wrapper = shallow(<DropdownImageTrigger/>);
        expect(wrapper).toMatchSnapshot()
    });

    // it('should render without crushing', () => {
    //     let wrapper = shallow(<NavBar/>);
    //     expect(wrapper).toMatchSnapshot()
    // });

});
