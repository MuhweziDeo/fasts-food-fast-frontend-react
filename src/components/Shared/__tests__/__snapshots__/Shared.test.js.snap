// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shared/> should render without crushing 1`] = `ShallowWrapper {}`;

exports[`<Shared/> should render without crushing 2`] = `ShallowWrapper {}`;
